src/main/java/net/dv8tion/jda/api/MessageBuilder.java

```diff
@@ -456,6 +461,251 @@ public MessageBuilder replaceLast(@Nonnull String target, @Nonnull String replac
+     */
+    @Nonnull
+    public MessageBuilder mention(@Nonnull IMentionable... mentions)
+    {
+        Checks.noneNull(mentions, "Mentions");
+
+        for (IMentionable mention : mentions)
+        {
+            if (mention instanceof User || mention instanceof Member)
+                mentionedUsers.add(mention.getId());
+            else if (mention instanceof Role)
+                mentionedRoles.add(mention.getId());
+        }
+        return this;
+    }
+
+    /**
+     * Adds the provided {@link IMentionable IMentionable} instance to the whitelist of mentions.
+     * <br>This will only affect instances of {@link User}, {@link Member}, and {@link Role}.
+     *
+     * <p>See {@link MessageAction#mention(IMentionable...)} for more details.
+     *
+     * @param  mentions
+     *         Whitelist of mentions to apply
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     *
+     * @see    #clearMentions()
+     * @see    MessageAction#mention(IMentionable...)
+     */
+    @Nonnull
+    public MessageBuilder mention(@Nonnull Collection<? extends IMentionable> mentions)
+    {
+        Checks.noneNull(mentions, "Mentions");
+        return mention(mentions.toArray(new IMentionable[0]));
+    }
+
+    /**
+     * Adds the provided {@link User Users} to the whitelist of mentions.
+     * <br>The provided list must only contain IDs of users.
+     *
+     * <p>See {@link MessageAction#mentionUsers(String...)} for more details.
+     *
+     * @param  users
+     *         Whitelist of user IDs to apply
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     *
+     * @see    #clearMentionedUsers()
+     * @see    MessageAction#mentionUsers(String...)
+     */
+    @Nonnull
+    public MessageBuilder mentionUsers(@Nonnull String... users)
+    {
+        Checks.noneNull(users, "Users");
+        Collections.addAll(mentionedUsers, users);
+        return this;
+    }
+
+    /**
+     * Adds the provided {@link Role Roles} to the whitelist of mentions.
+     * <br>The provided list must only contain IDs of roles.
+     *
+     * <p>See {@link MessageAction#mentionRoles(String...)} for more details.
+     *
+     * @param  roles
+     *         Whitelist of role IDs to apply
+     *
+     * @return The MessageBuilder instance. Useful for chaining.
+     *
+     * @see    #clearMentionedRoles()
+     * @see    MessageAction#mentionRoles(String...)
+     */
+    @Nonnull
+    public MessageBuilder mentionRoles(@Nonnull String... roles)
+    {
+        Checks.noneNull(roles, "Roles");
+        Collections.addAll(mentionedRoles, roles);
+        return this;
+    }
+
+    /**
+     * Adds the provided {@link User Users} to the whitelist of mentions.
+     * <br>The provided list must only contain IDs of users.
+     *
+     * <p>See {@link MessageAction#mentionUsers(long...)} for more details.
+     *
+     * @param  users
+     *         Whitelist of user IDs to apply
+     *
```

Comment:
Should this not have a `@throws IllegalArgumentException` considering `denyMentions(Message.MentionType...)` does?